cpp/src/arrow/python/arrow_to_python.cc

```diff
@@ -284,6 +284,7 @@ Status DeserializeObject(PyObject* context, const SerializedPyObject& obj, PyObj
                          PyObject** out) {
   PyAcquireGIL lock;
   PyDateTime_IMPORT;
+  import_pyarrow();
   return DeserializeList(context, *obj.batch->column(0), 0, obj.batch->num_rows(), base,
```

Comment:
I‘m confused that we need an `import_pyarrow` here. Can someone explain this better to me so I am more cautious about this problem? I would have assumed that we have imported all Arrow symbols in the case that any code in this `.cc` is called.

Reply:
I guess I also don't totally understand the function of this either (I understand why it's needed for other C++ libraries to be able to initiate the import, like NumPy's C import function), need to look more closely

Reply:
I have no idea. I got the idea while looking through the `SerializeObject` code (I assume that we also don't understand why it's needed there?). This fixes the segfault I was encountering.

Reply:
The reason why this is needed is that it imports the wrap_buffer and unwrap_buffer symbols from the cython extension which are needed in the buffer handling code of the Deserialize calls. In the serialialization codepath this already happens, so if we serialize something and then deserialize it, everything is fine. However, if we start a new python interpreter where the serialize codepath was never called and we try to deserialize an object that contains a buffer, it gives a segfault because import_pyarrow was never called.